src/main/webapp/javascript/user-auth.js

```diff
@@ -0,0 +1,96 @@
+var auth2;
+
+function getId() {
+  return auth2.currentUser.get().getId();
+}
+
+/**
+ * loads the auth2 library for the g-signin API, and then initializes
+ * the googleAuth object.
+ * passed as jquery parameter, executed by API script.
+ */
+function onStart() {
+  console.log("onstart");
+  gapi.load('auth2', initSigninV2);
+}
+
+/**
+ * initializes google authentication API and sets event listeners.
+ */
+function initSigninV2() {
+  auth2 = gapi.auth2.init({
+    client_id : '1034390229233-u07o0iaas2oql8l4jhe7fevpfsbrtv7n.apps.googleusercontent.com'
```

Comment:
Do you maybe want to make this client_id a constant at the top of the file with a comment on where it comes from?